core/src/main/java/org/jdbi/v3/Query.java

```diff
@@ -139,8 +139,8 @@
      * @see Handle#registerRowMapper(RowMapperFactory)
      * @see Handle#registerRowMapper(RowMapper)
      */
-    public Query mapTo(Type resultType) {
-        return this.map(new RegisteredRowMapper(resultType, config.mappingRegistry));
+    public <T> Query<T> mapTo(Type resultType) {
```

Comment:
Better to return `Query<?>` than to let shenanigans like this get past the compiler:

    Type type = int.class;
    String result = handle.createQuery("...").mapTo(type).findOnly();